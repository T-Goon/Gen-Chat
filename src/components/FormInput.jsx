import React from "react";
import { TextInput, Text, View, StyleSheet } from "react-native";

const FormInput = ({ placeholder, label, type, containerStyles, secureTextEntry }) => {

    return (
        <View style={{ ...styles.container, ...containerStyles }}>
            <Text>{label}</Text>

            <TextInput style={styles.input}
                placeholder={placeholder}
                keyboardType={type ? type : 'default'}
                secureTextEntry={secureTextEntry ? secureTextEntry : false} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        width: '100%',
        flexDirection: 'column',
        justifyContent: 'center'
    },
    input: {
        paddingHorizontal: 10,
        borderColor: "black",
        backgroundColor: 'hsl(0, 0%, 95%)',
        borderWidth: 2,
        borderRadius: 5,
    }
});

export default FormInput;