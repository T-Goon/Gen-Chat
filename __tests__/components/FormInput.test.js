import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import FormInput from '../../src/components/FormInput';

describe('<FormInput/ >', () => {
    it('Num Children', () => {
        const formInput = render(<FormInput containerStyles={{ paddingHorizontal: 15, marginVertical: 5, marginBottom: 20 }}
            label='Password:' placeholder='password' type='numeric' secureTextEntry={true} multiline={true} />);
        const json = formInput.toJSON();

        expect(json.children.length).toBe(2);
    });

    it('OnChange Prop', () => {
        const onChange = jest.fn();
        const { getByPlaceholderText } = render(<FormInput containerStyles={{ paddingHorizontal: 15, marginVertical: 5, marginBottom: 20 }}
            label='Password:' placeholder='password' secureTextEntry={false} onChangeText={onChange} multiline={false} />);

        fireEvent(getByPlaceholderText('password'), 'changeText', 'test');
        expect(onChange).toHaveBeenCalledWith('test');
    });

    it('Snapshot', () => {
        const formInputJson = render(<FormInput containerStyles={{ paddingHorizontal: 15, marginVertical: 5, marginBottom: 20 }}
            label='Password:' placeholder='password' secureTextEntry={true} multiline={false} />).toJSON();

        expect(formInputJson).toMatchSnapshot();
    });
})