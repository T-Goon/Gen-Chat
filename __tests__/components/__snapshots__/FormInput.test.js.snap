// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput/ > Snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 20,
      "marginVertical": 5,
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={Object {}}
  >
    Password:
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    multiline={false}
    onChangeText={null}
    placeholder="password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "hsl(0, 0%, 95%)",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 2,
        "paddingHorizontal": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
